sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/ComplexDataType.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.models;
+
+import com.azure.core.util.ExpandableStringEnum;
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+/**
+ * Simple DataType is used to create SimpleSearchField.
+ */
+public class ComplexDataType extends ExpandableStringEnum<ComplexDataType> {
```

Comment:
Don't feel great with using an expandable enum here, I feel this will lead to a bad direction. Instead could this be a simple model class where it is basically wrapping a string?

```java
public class ComplexDataType {
  private final String type;
  
  private ComplexDataType(String type) {
    this.type = type;
  }

  public static final ComplexDataType EDM_COMPLEX_TYPE = new ComplexDataType(DataType.EDM_COMPLEX_TYPE.toString());

  public static ComplexDataType COLLECTION = new ComplexDataType(String.format("Collection(%s)", EDM_COMPLEX_TYPE.type));

  public toString() {
    return type;
  }
}
```

Reply:
Yes, redundant of using extensible. Agreed on this